src/test/scala/chiselTests/ProbeSpec.scala

```diff
@@ -0,0 +1,349 @@
+
+          define(p, v)
+        },
+        Array("--throw-on-first-error")
+      )
+    }
+    exc.getMessage should be("Cannot define a probe on a non-equivalent type.")
+  }
+
+  "Probe of a probe type" should "fail" in {
+    val exc = intercept[chisel3.ChiselException] {
+      ChiselStage.emitCHIRRTL(
+        new RawModule {
+          val a = Output(RWProbe(Probe(Bool())))
+        },
+        Array("--throw-on-first-error")
+      )
+    }
+    exc.getMessage should be("Cannot probe a probe.")
+  }
+
+  "Probes of aggregates containing probes" should "fail" in {
+    val exc = intercept[chisel3.ChiselException] {
+      ChiselStage.emitCHIRRTL(
+        new RawModule {
+          val out = IO(Output(Probe(new Bundle {
+            val a = Probe(Bool())
+          })))
+        },
+        Array("--throw-on-first-error")
+      )
+    }
+    exc.getMessage should be("Cannot create a probe of an aggregate containing a probe.")
+  }
+
+  "Wire() of a probe" should "fail" in {
+    val exc = intercept[chisel3.ChiselException] {
+      ChiselStage.emitCHIRRTL(
+        new RawModule {
+          val w = Wire(Probe(Bool()))
+        },
+        Array("--throw-on-first-error")
+      )
+    }
+    exc.getMessage should be("Cannot make a wire of a Chisel type with a probe modifier.")
+  }
+
+  "WireInit of a probe" should "fail" in {
+    val exc = intercept[chisel3.ChiselException] {
+      ChiselStage.emitCHIRRTL(
+        new RawModule {
+          val w = WireInit(RWProbe(Bool()), false.B)
+        },
+        Array("--throw-on-first-error")
+      )
+    }
+    exc.getMessage should be("Cannot make a wire of a Chisel type with a probe modifier.")
+  }
+
+  "Reg() of a probe" should "fail" in {
+    val exc = intercept[chisel3.ChiselException] {
+      ChiselStage.emitCHIRRTL(
+        new RawModule {
+          val w = Reg(RWProbe(Bool()))
+        },
+        Array("--throw-on-first-error")
+      )
+    }
+    exc.getMessage should be("Cannot make a register of a Chisel type with a probe modifier.")
+  }
+
+  "RegInit of a probe" should "fail" in {
+    val exc = intercept[chisel3.ChiselException] {
+      ChiselStage.emitCHIRRTL(
+        new Module {
+          val w = RegInit(Probe(Bool()), false.B)
+        },
+        Array("--throw-on-first-error")
+      )
+    }
+    exc.getMessage should be("Cannot make a register of a Chisel type with a probe modifier.")
+  }
+
+  "Memories of probes" should "fail" in {
+    val exc = intercept[chisel3.ChiselException] {
+      ChiselStage.emitCHIRRTL(
+        new RawModule {
+          val mem = SyncReadMem(1024, RWProbe(Vec(4, UInt(32.W))))
+        },
+        Array("--throw-on-first-error")
+      )
+    }
+    exc.getMessage should be("Cannot make a Mem of a Chisel type with a probe modifier.")
+  }
+
+  // TODO probes of const type should work
+
+  // TODO writable probes of const type should fail
+
+  // TODO const of probe type should fail? --> put check in ConstSpec
+}
```

Comment:
These are some mega-terse tests. 💯

Reply:
These tests are awesome, thank you/great work!